File: ITapply/ITapply.Models/Requests/JobPostingInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Models.Requests
{
    public class JobPostingInsertRequest
    {
        [Required]
        public int EmployerId { get; set; }

        [Required]
        [StringLength(200)]
        public string Title { get; set; }

        [Required]
        [StringLength(10000)]
        public string Description { get; set; }

        [StringLength(5000)]
        public string Requirements { get; set; }

        [StringLength(3000)]
        public string Benefits { get; set; }

        [Required]
        public EmploymentType EmploymentType { get; set; }

        [Required]
        public ExperienceLevel ExperienceLevel { get; set; }

        public int? LocationId { get; set; }

        [Required]
        public Remote Remote { get; set; }

        [Range(0, int.MaxValue)]
        public int MinSalary { get; set; }

        [Range(0, int.MaxValue)]
        public int MaxSalary { get; set; }

        [Required]
        public DateTime ApplicationDeadline { get; set; }
        
        // List of skill IDs required for this job
        public List<int> SkillIds { get; set; } = new List<int>();
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/requests/job_posting_insert_request.dart
import 'package:itapply_mobile/models/enums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'job_posting_insert_request.g.dart';

@JsonSerializable()
class JobPostingInsertRequest {
  final int employerId;
  final String title;
  final String description;
  final String? requirements;
  final String? benefits;
  @JsonKey(toJson: _employmentTypeToJson)
  final EmploymentType employmentType;
  @JsonKey(toJson: _experienceLevelToJson)
  final ExperienceLevel experienceLevel;
  final int? locationId;
  @JsonKey(toJson: _remoteToJson)
  final Remote remote;
  final int? minSalary;
  final int? maxSalary;
  final DateTime applicationDeadline;
  final List<int>? skillIds;

  JobPostingInsertRequest({
    required this.employerId,
    required this.title,
    required this.description,
    this.requirements,
    this.benefits,
    required this.employmentType,
    required this.experienceLevel,
    this.locationId,
    required this.remote,
    this.minSalary,
    this.maxSalary,
    required this.applicationDeadline,
    this.skillIds,
  });

  factory JobPostingInsertRequest.fromJson(Map<String, dynamic> json) => _$JobPostingInsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$JobPostingInsertRequestToJson(this);
}

int _employmentTypeToJson(EmploymentType type) => type.index;
int _experienceLevelToJson(ExperienceLevel level) => level.index;
int _remoteToJson(Remote remote) => remote.index;
